### DISCUSSION TITLE
brew doctor telling me to git checkout master

### DISCUSSION THREAD

**Discussion by machale** [Homebrew/discussions#4120] [Everyday usage] [answered]
### Output of `brew config`

```shell
% brew config
HOMEBREW_VERSION: 3.6.19
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3e257890da96b4cc44f5933e3a54aee2b47549de
Last commit: 4 days ago
Branch: stable
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: d758bca1b5364d68643e6c64d281e52191f46780
Core tap last commit: 29 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_ANALYTICS: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit icelake
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.85.0 => /usr/bin/curl
macOS: 13.1-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
```

### Output of `brew doctor`

```shell
% brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew is not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo) checkout master"
```

### Description of issue

(I would have entered this in GitHub as an Issue / bug but the template for Homebrew bugs requires that my brew doctor be clean, and I don't want to take the `checkout master` action that would clean it.)

See the `brew doctor` output above. I am not developing formulae or anything, I'm just using brew to install and update formulae from others. 

IIUC, the master branch is only for developer mode in Homebrew. According to the [Manpage](https://docs.brew.sh/Manpage#developer-subcommand):

> developer [subcommand]
> Control Homebrew’s developer mode. When developer mode is enabled, brew update will update Homebrew to the latest commit on the master branch instead of the latest stable version along with some other behaviour changes.

I confirmed I’m not in developer mode:
```
% brew developer
Developer mode is disabled.
```

Furthermore, git confirms I’m on the stable branch:
```
% cd $(brew --repo)
% pwd
/usr/local/Homebrew
% git status
On branch stable
nothing to commit, working tree clean
```

Typically I keep my brew doctor clean and, in fact, it was clean just 3 hours earlier (approximately 10:20pm or 10:30pm US EST on January 17, 2023). Googling didn’t find anything obvious to me.

Is it correct that I should stay on the _stable_ branch and NOT run the suggested `checkout master` command? Was there a recent change in Homebrew that caused this warning?

EDIT: The trailing double quotes in the output of `brew doctor` is weird, as well.

**Comment by gromgit** [comment, marked as answer]
> Was there a recent change in Homebrew that caused this warning?

Yes, that was [my doing](https://github.com/Homebrew/brew/pull/14359). Your particular scenario [was just brought up](https://github.com/Homebrew/brew/pull/14359#issuecomment-1385988463), so I'll see how best to deal with it.

> EDIT: The trailing double quotes in the output of `brew doctor` is weird, as well.

That was me too, but [a fix is underway](https://github.com/Homebrew/brew/pull/14374).

**Comment by peterkoczan** [comment]
Thank you for flagging this, I am also having the same issue. Looking forward for the fix.

**Reply by gromgit** [reply]
The entire PR has been reverted (https://github.com/Homebrew/brew/pull/14380), so a `brew update` should fix it for everyone.

**Reply by pea2quared** [reply]
Yes, brew update did the trick, thanks!

**Comment by luckman212** [comment]
I was about to report this too, been bugging for a few days. I just saw that it was reverted but even after running `brew update && brew upgrade` several times, I'm still getting the warning.

I'm on branch `stable` — is that the correct place to be? Or do I need to reset back to `master`?

```
$ git -C $(brew --repo) branch
  main
  master
* stable
```

Sorry for being befuddled by this whole situation @gromgit

**Reply by gromgit** [reply]
Sorry, I spend all my time in developer mode, so I forgot about the `brew` release cadence. The fix will be available be available in the next version release.

In the meantime, `stable` is a perfectly cromulent branch to be on. :grinning:

**Comment by machale** [comment]
For future readers, this is fixed/reverted in release 3.6.20. https://github.com/Homebrew/brew/pull/14359#issuecomment-1397174174

`brew update` to get the change.

**Comment by audio-engineer** [comment]
Unfortunately, I am one of those that ran `git -C $(brew --repo) checkout master` when I got the `brew doctor` output mentioned in the description, and I have run `brew update` and `brew upgrade` since. Now `HOMEBREW_VERSION` is on `3.6.20-146-g4b3d64d` and I cannot switch back.
When I run `git -C $(brew --repo) checkout stable` and then `brew update && brew upgrade`, Homebrew automatically switches back to the `master` branch.

Any ideas how I can fix this?

**Reply by gromgit** [reply]
Try the following:
```
brew developer off
HOMEBREW_UPDATE_TO_TAG=1 brew update --force
```

**Reply by audio-engineer** [reply]
Fixed! Awesome! Thank you a lot

**Comment by Ender-Wang** [comment]
Hello, I somehow checked out to master branch and wanna checkout back to stable, but now there's no stable branch found:
```
* master
  remotes/origin/HEAD -> origin/master
  remotes/origin/master
```

I did remove .git in homebrew dir after migrating from my old m1 mac because `brew update` stops working after migration (seems obj tree is corrupted).

Now I cannot checkout to stable, or am I doing it wrong? Do `git checkout stable` yields 
```
git checkout stable
error: pathspec 'stable' did not match any file(s) known to git
```

**Reply by Ender-Wang** [reply]
Never mind, I did another installation of brew with /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)", not I am on stable. Thanks god.
